### ISSUE TITLE
NDM partitioned a cstor volume resulting in data corruption

### ISSUE THREAD

**Issue by akhilerm** [openebs/openebs#3221]
## Description

NDM created a partition on  a cstor volume that was being used by an application resulting in data corruption and thus application pod was not able to mount the volume.

## Expected Behavior

OpenEBS created iSCSI disks should be ignored by NDM, while creating partitions.

## Current Behavior

Have a multi-node cluster with application pod running on `master-01`
```
pi@k8s-master-01:~$ lsblk -fa
NAME        FSTYPE   LABEL       UUID                                 FSAVAIL FSUSE% MOUNTPOINT
loop0       squashfs                                                        0   100% /snap/core18/1883
loop1       squashfs                                                        0   100% /snap/core18/1888
loop2       squashfs                                                        0   100% /snap/lxd/16103
loop3       squashfs                                                        0   100% /snap/snapd/8543
loop4       squashfs                                                        0   100% /snap/lxd/16945
loop5       squashfs                                                        0   100% /snap/snapd/9287
loop6
loop7
loop8
loop9
loop10
.
.
.
loop255
mmcblk0
├─mmcblk0p1 vfat     system-boot B726-57E2                             155.8M    38% /boot/firmware
└─mmcblk0p2 ext4     writable    483efb12-d682-4daf-9b34-6e2f774b56f7   37.4G    32% /
```

#### What actually happened
When the master node was shutdown abruptly and rebooted, the application pod was having the following error in the pod describe:
```
Warning  FailedMount  44m (x9 over 3h38m)      kubelet, k8s-master-01  Unable to attach or mount volumes: unmounted volumes=[prometheus-x-prometheus-db], unattached volumes=[prometheus-x-prometheus-rulefiles-0 x-prometheus-token-wlk24 config config-out tls-assets prometheus-x-prometheus-db]: timed out waiting for the condition
  Warning  FailedMount  14m (x18 over 4h17m)     kubelet, k8s-master-01  Unable to attach or mount volumes: unmounted volumes=[prometheus-x-prometheus-db], unattached volumes=[config-out tls-assets prometheus-x-prometheus-db prometheus-x-prometheus-rulefiles-0 x-prometheus-token-wlk24 config]: timed out waiting for the condition
  Warning  FailedMount  4m11s (x194 over 4h26m)  kubelet, k8s-master-01  (combined from similar events): MountVolume.MountDevice failed for volume "pvc-5735a46f-416d-40d5-8a41-e55122dda3df" : mount failed: exit status 32
Mounting command: systemd-run
Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/10.102.4.25:3260-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0 --scope -- mount -t ext4 -o defaults /dev/disk/by-path/ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0 /var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/10.102.4.25:3260-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
Output: Running scope as unit: run-rc00c0551102e411598debcae9f6a9804.scope
mount: /var/lib/kubelet/plugins/kubernetes.io/iscsi/iface-default/10.102.4.25:3260-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0: wrong fs type, bad option, bad superblock on /dev/sda, missing codepage or helper program, or other error.
```

The cstor volumes  are in healthy state
```
❯ kubectl get cstorvolume -n storage
NAME                                       STATUS    AGE    CAPACITY
pvc-5735a46f-416d-40d5-8a41-e55122dda3df   Healthy   3d9h   10Gi

❯ kubectl get cvr -n storage
NAME                                                              USED    ALLOCATED   STATUS    AGE
pvc-5735a46f-416d-40d5-8a41-e55122dda3df-cstor-sparse-pool-a59p   4.45G   2.35G       Healthy   3d9h
pvc-5735a46f-416d-40d5-8a41-e55122dda3df-cstor-sparse-pool-wl0q   4.45G   2.35G       Healthy   3d9h
pvc-5735a46f-416d-40d5-8a41-e55122dda3df-cstor-sparse-pool-zs99   4.45G   2.35G       Healthy   3d9h
```

iscsi adm output
```
❯ sudo iscsiadm -m session -P 3
iSCSI Transport Class version 2.0-870
version 2.0-874
Target: iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df (non-flash)
	Current Portal: 10.102.4.25:3260,1
	Persistent Portal: 10.102.4.25:3260,1
		**********
		Interface:
		**********
		Iface Name: default
		Iface Transport: tcp
		Iface Initiatorname: iqn.1993-08.org.debian:01:d68048bab763
		Iface IPaddress: 192.168.1.121
		Iface HWaddress: <empty>
		Iface Netdev: <empty>
		SID: 1
		iSCSI Connection State: LOGGED IN
		iSCSI Session State: LOGGED_IN
		Internal iscsid Session State: NO CHANGE
		*********
		Timeouts:
		*********
		Recovery Timeout: 120
		Target Reset Timeout: 30
		LUN Reset Timeout: 30
		Abort Timeout: 15
		*****
		CHAP:
		*****
		username: <empty>
		password: ********
		username_in: <empty>
		password_in: ********
		************************
		Negotiated iSCSI params:
		************************
		HeaderDigest: None
		DataDigest: None
		MaxRecvDataSegmentLength: 262144
		MaxXmitDataSegmentLength: 262144
		FirstBurstLength: 262144
		MaxBurstLength: 1048576
		ImmediateData: Yes
		InitialR2T: Yes
		MaxOutstandingR2T: 1
		************************
		Attached SCSI devices:
		************************
		Host Number: 0	State: running
		scsi0 Channel 00 Id 0 Lun: 0
			Attached scsi disk sda		State: running
```

The target pod was running on a different node (master-03), which was not shutdown.

`lsblk` output shows a partition on the scsi disk, instead of a filesystem
```
pi@k8s-master-01:~$ sudo lsblk -fa /dev/sda
NAME   FSTYPE LABEL UUID FSAVAIL FSUSE% MOUNTPOINT
sda
└─sda1

pi@k8s-master-01:~$ udevadm info /dev/sda
P: /devices/platform/host0/session1/target0:0:0/0:0:0:0/block/sda
N: sda
L: 0
S: disk/by-id/scsi-3300000000e960cac
S: disk/by-id/scsi-SOpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
S: disk/by-id/scsi-1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
S: disk/by-id/wwn-0x300000000e960cac
S: disk/by-path/ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: DEVPATH=/devices/platform/host0/session1/target0:0:0/0:0:0:0/block/sda
E: DEVNAME=/dev/sda
E: DEVTYPE=disk
E: MAJOR=8
E: MINOR=0
E: SUBSYSTEM=block
E: USEC_INITIALIZED=127865380
E: SCSI_TPGS=0
E: SCSI_TYPE=disk
E: SCSI_VENDOR=OpenEBS
E: SCSI_VENDOR_ENC=OpenEBS\x20
E: SCSI_MODEL=iscsi
E: SCSI_MODEL_ENC=iscsi\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: SCSI_REVISION=0
E: ID_SCSI=1
E: ID_SCSI_INQUIRY=1
E: ID_VENDOR=OpenEBS
E: ID_VENDOR_ENC=OpenEBS\x20
E: ID_MODEL=iscsi
E: ID_MODEL_ENC=iscsi\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: ID_REVISION=0
E: ID_TYPE=disk
E: SCSI_IDENT_SERIAL=04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: SCSI_IDENT_LUN_NAA_LOCAL=300000000e960cac
E: SCSI_IDENT_LUN_T10=OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: SCSI_IDENT_TARGET_NAME=iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df
E: SCSI_IDENT_PORT_NAME=iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df,t,0x0001
E: SCSI_IDENT_PORT_RELATIVE=1
E: SCSI_IDENT_PORT_TARGET_PORT_GROUP=0x1
E: ID_WWN=0x300000000e960cac
E: ID_WWN_WITH_EXTENSION=0x300000000e960cac
E: ID_BUS=scsi
E: ID_SERIAL=1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: ID_SERIAL_SHORT=OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: MPATH_SBIN_PATH=/sbin
E: DM_MULTIPATH_DEVICE_PATH=0
E: ID_PATH=ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: ID_PATH_TAG=ip-10_102_4_25_3260-iscsi-iqn_2016-09_com_openebs_cstor_pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: ID_PART_TABLE_UUID=b9d36056-c434-4f6e-92ee-dfbb33e28f79
E: ID_PART_TABLE_TYPE=gpt
E: DEVLINKS=/dev/disk/by-id/scsi-3300000000e960cac /dev/disk/by-id/scsi-SOpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29 /dev/disk/by-id/scsi-1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29 /dev/disk/by-id/wwn-0x300000000e960cac /dev/disk/by-path/ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: TAGS=:systemd:

pi@k8s-master-01:~$ udevadm info /dev/sda1
P: /devices/platform/host0/session1/target0:0:0/0:0:0:0/block/sda/sda1
N: sda1
L: 0
S: disk/by-partuuid/a06bcbcc-21f2-4839-b8bf-16a781282259
S: disk/by-id/wwn-0x300000000e960cac-part1
S: disk/by-path/ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0-part1
S: disk/by-id/scsi-SOpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29-part1
S: disk/by-id/scsi-1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29-part1
S: disk/by-id/scsi-3300000000e960cac-part1
E: DEVPATH=/devices/platform/host0/session1/target0:0:0/0:0:0:0/block/sda/sda1
E: DEVNAME=/dev/sda1
E: DEVTYPE=partition
E: PARTN=1
E: MAJOR=8
E: MINOR=1
E: SUBSYSTEM=block
E: USEC_INITIALIZED=17295719021
E: SCSI_TPGS=0
E: SCSI_TYPE=disk
E: SCSI_VENDOR=OpenEBS
E: SCSI_VENDOR_ENC=OpenEBS\x20
E: SCSI_MODEL=iscsi
E: SCSI_MODEL_ENC=iscsi\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: SCSI_REVISION=0
E: SCSI_IDENT_SERIAL=04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: SCSI_IDENT_LUN_NAA_LOCAL=300000000e960cac
E: SCSI_IDENT_LUN_T10=OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: SCSI_IDENT_TARGET_NAME=iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df
E: SCSI_IDENT_PORT_NAME=iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df,t,0x0001
E: SCSI_IDENT_PORT_RELATIVE=1
E: SCSI_IDENT_PORT_TARGET_PORT_GROUP=0x1
E: ID_SCSI=1
E: ID_VENDOR=OpenEBS
E: ID_VENDOR_ENC=OpenEBS\x20
E: ID_MODEL=iscsi
E: ID_MODEL_ENC=iscsi\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20\x20
E: ID_REVISION=0
E: ID_TYPE=disk
E: ID_WWN=0x300000000e960cac
E: ID_WWN_WITH_EXTENSION=0x300000000e960cac
E: ID_BUS=scsi
E: ID_SERIAL=1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: ID_SERIAL_SHORT=OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29
E: DM_MULTIPATH_DEVICE_PATH=0
E: ID_SCSI_INQUIRY=1
E: ID_PATH=ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: ID_PATH_TAG=ip-10_102_4_25_3260-iscsi-iqn_2016-09_com_openebs_cstor_pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0
E: ID_PART_TABLE_UUID=b9d36056-c434-4f6e-92ee-dfbb33e28f79
E: ID_PART_TABLE_TYPE=gpt
E: ID_PART_ENTRY_SCHEME=gpt
E: ID_PART_ENTRY_UUID=a06bcbcc-21f2-4839-b8bf-16a781282259
E: ID_PART_ENTRY_TYPE=0fc63daf-8483-4772-8e79-3d69d8477de4
E: ID_PART_ENTRY_NUMBER=1
E: ID_PART_ENTRY_OFFSET=2048
E: ID_PART_ENTRY_SIZE=20969439
E: ID_PART_ENTRY_DISK=8:0
E: DEVLINKS=/dev/disk/by-partuuid/a06bcbcc-21f2-4839-b8bf-16a781282259 /dev/disk/by-id/wwn-0x300000000e960cac-part1 /dev/disk/by-path/ip-10.102.4.25:3260-iscsi-iqn.2016-09.com.openebs.cstor:pvc-5735a46f-416d-40d5-8a41-e55122dda3df-lun-0-part1 /dev/disk/by-id/scsi-SOpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29-part1 /dev/disk/by-id/scsi-1OpenEBS_iscsi_04c24d7f-c2ff-4b5f-a55c-4b1c2a29-part1 /dev/disk/by-id/scsi-3300000000e960cac-part1
E: TAGS=:systemd:
```

BlockDevices as discovered by NDM
```
❯ kubectl get bd -n storage -o wide

NAME                                           NODENAME        PATH                                   FSTYPE   SIZE          CLAIMSTATE   STATUS     AGE
blockdevice-08da59065b1b68705808e412b19439a6   k8s-master-01   /dev/sda1                                       10736352768   Unclaimed    Inactive   7h37m
sparse-55fd7ec10d680eb096399e08a2fb11bc        k8s-master-03   /var/openebs/sparse/0-ndm-sparse.img            10737418240   Claimed      Active     3d12h
sparse-775fd37c446bc03ff821b8c487da95b7        k8s-master-01   /var/openebs/sparse/0-ndm-sparse.img            10737418240   Claimed      Active     3d12h
sparse-859ecd67c32c17331ad1ad0abde080bb        k8s-worker-01   /var/openebs/sparse/0-ndm-sparse.img            10737418240   Unclaimed    Active     3d12h
sparse-963b99361088e0a766d82d6fb3cab831        k8s-master-02   /var/openebs/sparse/0-ndm-sparse.img            10737418240   Claimed      Active     3d12h
```
```
❯ kubectl get bd -n storage -o yaml
apiVersion: v1
items:
- apiVersion: openebs.io/v1alpha1
  kind: BlockDevice
  metadata:
    annotations:
      internal.openebs.io/uuid-scheme: gpt
    creationTimestamp: "2020-10-04T03:59:54Z"
    generation: 2
    labels:
      kubernetes.io/hostname: k8s-master-01
      ndm.io/blockdevice-type: blockdevice
      ndm.io/managed: "true"
    managedFields:
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:annotations:
            .: {}
            f:internal.openebs.io/uuid-scheme: {}
          f:labels:
            .: {}
            f:kubernetes.io/hostname: {}
            f:ndm.io/blockdevice-type: {}
            f:ndm.io/managed: {}
        f:spec:
          .: {}
          f:capacity:
            .: {}
            f:logicalSectorSize: {}
            f:physicalSectorSize: {}
            f:storage: {}
          f:details:
            .: {}
            f:compliance: {}
            f:deviceType: {}
            f:driveType: {}
            f:firmwareRevision: {}
            f:hardwareSectorSize: {}
            f:logicalBlockSize: {}
            f:model: {}
            f:physicalBlockSize: {}
            f:serial: {}
            f:vendor: {}
          f:devlinks: {}
          f:filesystem: {}
          f:nodeAttributes:
            .: {}
            f:nodeName: {}
          f:partitioned: {}
          f:path: {}
        f:status:
          .: {}
          f:claimState: {}
          f:state: {}
      manager: ndm
      operation: Update
      time: "2020-10-04T03:59:54Z"
    name: blockdevice-08da59065b1b68705808e412b19439a6
    namespace: storage
    resourceVersion: "2466340"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/storage/blockdevices/blockdevice-08da59065b1b68705808e412b19439a6
    uid: a676b887-a662-4957-8aba-8e55878116ce
  spec:
    capacity:
      logicalSectorSize: 512
      physicalSectorSize: 0
      storage: 10736352768
    details:
      compliance: SPC-3
      deviceType: partition
      driveType: ""
      firmwareRevision: '0   '
      hardwareSectorSize: 0
      logicalBlockSize: 512
      model: ""
      physicalBlockSize: 0
      serial: ""
      vendor: ""
    devlinks: []
    filesystem: {}
    nodeAttributes:
      nodeName: k8s-master-01
    partitioned: "No"
    path: /dev/sda1
  status:
    claimState: Unclaimed
    state: Inactive
- apiVersion: openebs.io/v1alpha1
  kind: BlockDevice
  metadata:
    creationTimestamp: "2020-09-30T22:54:43Z"
    finalizers:
    - openebs.io/bd-protection
    generation: 2
    labels:
      kubernetes.io/hostname: k8s-master-03
      ndm.io/blockdevice-type: blockdevice
      ndm.io/managed: "true"
    managedFields:
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:labels:
            .: {}
            f:kubernetes.io/hostname: {}
            f:ndm.io/blockdevice-type: {}
            f:ndm.io/managed: {}
        f:spec:
          .: {}
          f:capacity:
            .: {}
            f:logicalSectorSize: {}
            f:physicalSectorSize: {}
            f:storage: {}
          f:details:
            .: {}
            f:compliance: {}
            f:deviceType: {}
            f:driveType: {}
            f:firmwareRevision: {}
            f:hardwareSectorSize: {}
            f:logicalBlockSize: {}
            f:model: {}
            f:physicalBlockSize: {}
            f:serial: {}
            f:vendor: {}
          f:devlinks: {}
          f:filesystem: {}
          f:nodeAttributes:
            .: {}
            f:nodeName: {}
          f:partitioned: {}
          f:path: {}
        f:status:
          .: {}
          f:state: {}
      manager: ndm
      operation: Update
      time: "2020-09-30T22:54:43Z"
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers: {}
        f:spec:
          f:claimRef:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
            f:namespace: {}
            f:resourceVersion: {}
            f:uid: {}
        f:status:
          f:claimState: {}
      manager: ndo
      operation: Update
      time: "2020-09-30T22:55:12Z"
    name: sparse-55fd7ec10d680eb096399e08a2fb11bc
    namespace: storage
    resourceVersion: "5561"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/storage/blockdevices/sparse-55fd7ec10d680eb096399e08a2fb11bc
    uid: 326192fb-df4d-4516-ae03-c7d8807976fe
  spec:
    capacity:
      logicalSectorSize: 0
      physicalSectorSize: 0
      storage: 10737418240
    claimRef:
      apiVersion: openebs.io/v1alpha1
      kind: BlockDeviceClaim
      name: bdc-326192fb-df4d-4516-ae03-c7d8807976fe
      namespace: storage
      resourceVersion: "5558"
      uid: b83d8dae-7840-4f6e-b224-51c37a157f8d
    details:
      compliance: ""
      deviceType: sparse
      driveType: ""
      firmwareRevision: ""
      hardwareSectorSize: 0
      logicalBlockSize: 0
      model: ""
      physicalBlockSize: 0
      serial: ""
      vendor: ""
    devlinks: []
    filesystem: {}
    nodeAttributes:
      nodeName: k8s-master-03
    partitioned: "No"
    path: /var/openebs/sparse/0-ndm-sparse.img
  status:
    claimState: Claimed
    state: Active
- apiVersion: openebs.io/v1alpha1
  kind: BlockDevice
  metadata:
    creationTimestamp: "2020-09-30T22:54:39Z"
    finalizers:
    - openebs.io/bd-protection
    generation: 2
    labels:
      kubernetes.io/hostname: k8s-master-01
      ndm.io/blockdevice-type: blockdevice
      ndm.io/managed: "true"
    managedFields:
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:labels:
            .: {}
            f:kubernetes.io/hostname: {}
            f:ndm.io/blockdevice-type: {}
            f:ndm.io/managed: {}
        f:spec:
          .: {}
          f:capacity:
            .: {}
            f:logicalSectorSize: {}
            f:physicalSectorSize: {}
            f:storage: {}
          f:details:
            .: {}
            f:compliance: {}
            f:deviceType: {}
            f:driveType: {}
            f:firmwareRevision: {}
            f:hardwareSectorSize: {}
            f:logicalBlockSize: {}
            f:model: {}
            f:physicalBlockSize: {}
            f:serial: {}
            f:vendor: {}
          f:devlinks: {}
          f:filesystem: {}
          f:nodeAttributes:
            .: {}
            f:nodeName: {}
          f:partitioned: {}
          f:path: {}
        f:status:
          .: {}
          f:state: {}
      manager: ndm
      operation: Update
      time: "2020-09-30T22:54:39Z"
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers: {}
        f:spec:
          f:claimRef:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
            f:namespace: {}
            f:resourceVersion: {}
            f:uid: {}
        f:status:
          f:claimState: {}
      manager: ndo
      operation: Update
      time: "2020-09-30T22:55:11Z"
    name: sparse-775fd37c446bc03ff821b8c487da95b7
    namespace: storage
    resourceVersion: "5538"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/storage/blockdevices/sparse-775fd37c446bc03ff821b8c487da95b7
    uid: 93965c2b-10a0-4a92-a026-48e145012600
  spec:
    capacity:
      logicalSectorSize: 0
      physicalSectorSize: 0
      storage: 10737418240
    claimRef:
      apiVersion: openebs.io/v1alpha1
      kind: BlockDeviceClaim
      name: bdc-93965c2b-10a0-4a92-a026-48e145012600
      namespace: storage
      resourceVersion: "5537"
      uid: e1cbdefc-ffde-455f-8e6b-616f54c28dbf
    details:
      compliance: ""
      deviceType: sparse
      driveType: ""
      firmwareRevision: ""
      hardwareSectorSize: 0
      logicalBlockSize: 0
      model: ""
      physicalBlockSize: 0
      serial: ""
      vendor: ""
    devlinks: []
    filesystem: {}
    nodeAttributes:
      nodeName: k8s-master-01
    partitioned: "No"
    path: /var/openebs/sparse/0-ndm-sparse.img
  status:
    claimState: Claimed
    state: Active
- apiVersion: openebs.io/v1alpha1
  kind: BlockDevice
  metadata:
    creationTimestamp: "2020-09-30T22:54:42Z"
    generation: 1
    labels:
      kubernetes.io/hostname: k8s-worker-01
      ndm.io/blockdevice-type: blockdevice
      ndm.io/managed: "true"
    managedFields:
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:labels:
            .: {}
            f:kubernetes.io/hostname: {}
            f:ndm.io/blockdevice-type: {}
            f:ndm.io/managed: {}
        f:spec:
          .: {}
          f:capacity:
            .: {}
            f:logicalSectorSize: {}
            f:physicalSectorSize: {}
            f:storage: {}
          f:details:
            .: {}
            f:compliance: {}
            f:deviceType: {}
            f:driveType: {}
            f:firmwareRevision: {}
            f:hardwareSectorSize: {}
            f:logicalBlockSize: {}
            f:model: {}
            f:physicalBlockSize: {}
            f:serial: {}
            f:vendor: {}
          f:devlinks: {}
          f:filesystem: {}
          f:nodeAttributes:
            .: {}
            f:nodeName: {}
          f:partitioned: {}
          f:path: {}
        f:status:
          .: {}
          f:claimState: {}
          f:state: {}
      manager: ndm
      operation: Update
      time: "2020-09-30T22:54:42Z"
    name: sparse-859ecd67c32c17331ad1ad0abde080bb
    namespace: storage
    resourceVersion: "5083"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/storage/blockdevices/sparse-859ecd67c32c17331ad1ad0abde080bb
    uid: 3e84cf4f-d854-4493-ba98-726f9eb72925
  spec:
    capacity:
      logicalSectorSize: 0
      physicalSectorSize: 0
      storage: 10737418240
    details:
      compliance: ""
      deviceType: sparse
      driveType: ""
      firmwareRevision: ""
      hardwareSectorSize: 0
      logicalBlockSize: 0
      model: ""
      physicalBlockSize: 0
      serial: ""
      vendor: ""
    devlinks: []
    filesystem: {}
    nodeAttributes:
      nodeName: k8s-worker-01
    partitioned: "No"
    path: /var/openebs/sparse/0-ndm-sparse.img
  status:
    claimState: Unclaimed
    state: Active
- apiVersion: openebs.io/v1alpha1
  kind: BlockDevice
  metadata:
    creationTimestamp: "2020-09-30T22:54:43Z"
    finalizers:
    - openebs.io/bd-protection
    generation: 2
    labels:
      kubernetes.io/hostname: k8s-master-02
      ndm.io/blockdevice-type: blockdevice
      ndm.io/managed: "true"
    managedFields:
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:labels:
            .: {}
            f:kubernetes.io/hostname: {}
            f:ndm.io/blockdevice-type: {}
            f:ndm.io/managed: {}
        f:spec:
          .: {}
          f:capacity:
            .: {}
            f:logicalSectorSize: {}
            f:physicalSectorSize: {}
            f:storage: {}
          f:details:
            .: {}
            f:compliance: {}
            f:deviceType: {}
            f:driveType: {}
            f:firmwareRevision: {}
            f:hardwareSectorSize: {}
            f:logicalBlockSize: {}
            f:model: {}
            f:physicalBlockSize: {}
            f:serial: {}
            f:vendor: {}
          f:devlinks: {}
          f:filesystem: {}
          f:nodeAttributes:
            .: {}
            f:nodeName: {}
          f:partitioned: {}
          f:path: {}
        f:status:
          .: {}
          f:state: {}
      manager: ndm
      operation: Update
      time: "2020-09-30T22:54:43Z"
    - apiVersion: openebs.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:metadata:
          f:finalizers: {}
        f:spec:
          f:claimRef:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
            f:namespace: {}
            f:resourceVersion: {}
            f:uid: {}
        f:status:
          f:claimState: {}
      manager: ndo
      operation: Update
      time: "2020-09-30T22:55:12Z"
    name: sparse-963b99361088e0a766d82d6fb3cab831
    namespace: storage
    resourceVersion: "5555"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/storage/blockdevices/sparse-963b99361088e0a766d82d6fb3cab831
    uid: 45bb91f3-1e0b-468f-bd15-9a022a192c72
  spec:
    capacity:
      logicalSectorSize: 0
      physicalSectorSize: 0
      storage: 10737418240
    claimRef:
      apiVersion: openebs.io/v1alpha1
      kind: BlockDeviceClaim
      name: bdc-45bb91f3-1e0b-468f-bd15-9a022a192c72
      namespace: storage
      resourceVersion: "5553"
      uid: 192c8e0d-7dc5-4fb2-8527-9e14cc1da6c4
    details:
      compliance: ""
      deviceType: sparse
      driveType: ""
      firmwareRevision: ""
      hardwareSectorSize: 0
      logicalBlockSize: 0
      model: ""
      physicalBlockSize: 0
      serial: ""
      vendor: ""
    devlinks: []
    filesystem: {}
    nodeAttributes:
      nodeName: k8s-master-02
    partitioned: "No"
    path: /var/openebs/sparse/0-ndm-sparse.img
  status:
    claimState: Claimed
    state: Active
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

NDM pod [logs](https://github.com/openebs/openebs/files/5335739/ndm_pod_logs.txt) from master node

From the ndm logs and blockdevice resource, it seems NDM has written the partition on the cstor volume. The vendor filter and check for WWN didn't work, though these fields were present when later checked via udev

## Steps to Reproduce

Was not able to reproduce again with node shutdown.

## Your Environment

* `kubectl get nodes`:
* `kubectl get pods --all-namespaces`:
* `kubectl get services`:
* `kubectl get sc`:
* `kubectl get pv`:
* `kubectl get pvc`:
* OS (from `/etc/os-release`):
* Kernel (from `uname -a`):
* Install tools:
* Others:

**Comment by akhilerm** [comment]
@xUnholy , can you update the issue with more info on your cluster setup, like type of machines used , OS and kernel version.

**Comment by akhilerm** [comment]
A fix to prevent partitions from being created, by doing an additional filesystem check has been merged in openebs/node-disk-manager#496
